backend/src/packages/trucks/truck.repository.ts

```diff
@@ -0,0 +1,170 @@
+import { eq, ilike, or } from 'drizzle-orm';
+
+import { type IRepository } from '~/libs/interfaces/interfaces.js';
+import { type IDatabase } from '~/libs/packages/database/database.js';
+import { type DatabaseSchema } from '~/libs/packages/database/schema/schema.js';
+
+import { type TruckEntity as TruckEntityT } from './libs/types/types.js';
+import { TruckEntity } from './truck.entity.js';
+
+type NewTruckData = Omit<TruckEntityT, 'id' | 'createdAt' | 'updatedAt'>;
+
+class TruckRepository implements IRepository {
+  private db: Pick<IDatabase, 'driver'>;
+
+  private trucksSchema: DatabaseSchema['trucks'];
+
+  private usersTrucksSchema: DatabaseSchema['usersTrucks'];
+
+  public constructor(
+    database: Pick<IDatabase, 'driver'>,
+    trucksSchema: DatabaseSchema['trucks'],
+    usersTrucksSchema: DatabaseSchema['usersTrucks'],
+  ) {
+    this.db = database;
+    this.trucksSchema = trucksSchema;
+    this.usersTrucksSchema = usersTrucksSchema;
+  }
+
+  public async find(id: number): Promise<TruckEntity | null> {
+    const result: TruckEntityT[] = await this.db
+      .driver()
+      .select()
+      .from(this.trucksSchema)
+      .where(eq(this.trucksSchema.id, id));
+
+    return result.length === 0 ? null : TruckEntity.initialize(result[0]);
```

Comment:
```suggestion
    const [truck = null]: TruckEntityT[] = await this.db
      .driver()
      .select()
      .from(this.trucksSchema)
      .where(eq(this.trucksSchema.id, id));

    return truck ? TruckEntity.initialize(truck) : null;
```
Don't need to fix this now, because we decided to change approach. Just FYI